File: app/src/main/java/com/raju/javabaseproject/data/utils/NetworkUtil.java
package com.raju.javabaseproject.data.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import javax.inject.Inject;
import javax.inject.Singleton;

/**
 * Created by Rajashekhar Vanahalli on 05/04/18.
 */

@Singleton
public class NetworkUtil {

    @Inject
    NetworkUtil(Context context) {
        this.connectivity = new Connectivity(context);
        connectionFast = connectivity.isConnectedFast();
    }

    public boolean update() {
        connectionFast = connectivity.isConnectedFast();
        return isConnectionFast();
    }

    public boolean isConnectionFast() {
        return connectionFast;
    }

    public boolean isConnected() {
        return connectivity.isConnected();
    }

    public static boolean isConnected(Context context) {
        ConnectivityManager connectivityManager = (ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo netInfo = connectivityManager.getActiveNetworkInfo();
        return (netInfo != null && netInfo.isConnected());
    }

    private final Connectivity connectivity;
    private boolean connectionFast = false;
}

File: app/src/main/java/com/raju/javabaseproject/data/utils/NetworkUtil.kt
package com.raju.javabaseproject.data.utils

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

import javax.inject.Inject
import javax.inject.Singleton


@Singleton
class NetworkUtil @Inject
internal constructor(context: Context) {

    val isConnected: Boolean
        get() = connectivity.isConnected

    private val connectivity: Connectivity
    var isConnectionFast = false
        private set

    init {
        this.connectivity = Connectivity(context)
        isConnectionFast = connectivity.isConnectedFast
    }

    fun update(): Boolean {
        isConnectionFast = connectivity.isConnectedFast
        return isConnectionFast
    }

    companion object {

        fun isConnected(context: Context): Boolean {
            val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
            val netInfo = connectivityManager.activeNetworkInfo
            return netInfo != null && netInfo.isConnected
        }
    }
}

